### ISSUE TITLE
Box-shadow used as border should be inset. Otherwise they may be cropped

### ISSUE THREAD

**Issue by BertrandPodevin** [LuccaSA/lucca-front#2647]
IF a component use a `box-shadow: 0 0 0 1px #color;` as a border, this shadow should also be defined as `inset`.
Otherwise the shadow overflow it's box and could be cropped by it's parents.

Example with a `callout` : the laterals shadows are cropped.
![image](https://github.com/LuccaSA/lucca-front/assets/65592565/7611ad38-7a10-47af-9313-23c7585b6f15)

That would'nt be the case with an `inset` shadow :
![image](https://github.com/LuccaSA/lucca-front/assets/65592565/0106b777-77b6-4e6a-a107-8176b79ff162)

**Comment by BertrandPodevin** [comment]
@jeremie-lucca Avoiding using overflow is not always possibility :/

Which leaves 4 options :
- Not using this element at all :(
- Keep a super ugly cropped element :(
- Adding artifical `margin` around the element to keep it uncropped but break alignments with other elements
- Overriding the box shadow, but because there is not `box-shadow-inset` property in css we need to override the whole box-shadows, and for every states (:hover, :focus...) 

An element designed to take `100%` of it's parent width should not casualy take `100% + 2px` without offering a fallback
There should be at least a `mod` or a `css vars` to allow a clean override when needed.

Ex `box-shadow: var(--component-box-shadow-inset, ) 0px 0px 15px 0px #000000;`
which can be overrided easily  `.myComponent {--component-box-shadow-inset: inset;}`

**Comment by jeremie-lucca** [comment]
@BertrandPodevin Could you share some examples/mockups where you cannot bypass hidden overflow ?
May concrete examples could convince Clément to use inside borders. If not, we'll consider fallbacks 🙂

**Comment by BertrandPodevin** [comment]
I don't think that there are designs or mockup where you need to combine outset shadow component + overflow.

On the other hand this change (border -> shadow) is breaking for existing screens build without this technical constraint in mind. Some TA screens and our pwa are exposed now to this. 

A simple fallback will do the trick.

**Comment by vvalentin-lucca** [comment]
The same problem can arise when it comes to displaying the outline for keyboard navigation.

You can get away with [a negative margin combined with padding of the same value](https://codepen.io/vincent-valentin/pen/PorPNLP/e0d051c8aa76b9758406eb25d94e8a01?editors=1100), if necessary.
